app/src/features/apiClient/APIClientManager/workerScripts.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-new-func */
+
+import { EnvironmentVariables } from "backend/environment/types";
+
+export const requestWorkerFunction = function (e: MessageEvent) {
+  const { script, request, currentVariables } = e.data;
+
+  const variableSaverQueue: EnvironmentVariables = {};
+  const variableRemoverQueue: string[] = [];
+
+  const sandbox = {
+    rq: {
+      request,
+      environment: {
+        set: (key: string, value: any) => {
+          variableSaverQueue[key] = {
+            localValue: value,
+            syncValue: value,
+            type: typeof value,
+          };
+        },
+        get: (key: string) => {
+          const variable = currentVariables[key];
+          return variable?.localValue || variable?.syncValue;
```

Comment:
We should decide this logic in the `APIManager` only. APIManager should send only the final env values to here and let the variables processor handle the rest (type conversion, fallback values etc) in a single place.

```
{
    [key: string]: string|number...,
}
```

Reply:
This is still not resolved yet.
Please make sure to incorporate this in refractoring